plugins/jokes.py

```diff
@@ -1,95 +1,98 @@
-        yo_momma = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "do_it.txt"), encoding="utf-8") as f:
-        do_it = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "puns.txt"), encoding="utf-8") as f:
-        pun = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "confucious.txt"), encoding="utf-8") as f:
-        confucious = [line.strip() for line in f.readlines() if not line.startswith("//")]
+    global yo_momma, do_it, puns, confucious_say, one_liner, wise_quotes, book_puns, lawyer_jokes, kero_sayings
 
-    with codecs.open(os.path.join(bot.data_dir, "one_liners.txt"), encoding="utf-8") as f:
-        one_liner = [line.strip() for line in f.readlines() if not line.startswith("//")]
+    data_directory = Path(bot.data_dir)
 
-    with codecs.open(os.path.join(bot.data_dir, "wisdom.txt"), encoding="utf-8") as f:
-        wisdom = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "book_puns.txt"), encoding="utf-8") as f:
-        book_puns = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "lawyerjoke.txt"), encoding="utf-8") as f:
-        lawyerjoke = [line.strip() for line in f.readlines() if not line.startswith("//")]
-
-    with codecs.open(os.path.join(bot.data_dir, "kero.txt"), encoding="utf-8") as f:
-        kero_sayings = [line.strip() for line in f.readlines() if not line.startswith("//")]
+    yo_momma = load_joke_file(data_directory / 'yo_momma.txt')
+    do_it = load_joke_file(data_directory / 'do_it.txt')
+    puns = load_joke_file(data_directory / 'puns.txt')
+    confucious_say = load_joke_file(data_directory / 'confucious.txt')
+    one_liner = load_joke_file(data_directory / 'one_liners.txt')
+    wise_quotes = load_joke_file(data_directory / 'wisdom.txt')
+    book_puns = load_joke_file(data_directory / 'book_puns.txt')
+    lawyer_jokes = load_joke_file(data_directory / 'lawyerjoke.txt')
+    kero_sayings = load_joke_file(data_directory / 'kero.txt')
 
 
 @hook.command()
-def yomomma(text):
-    """<nick> - tells a yo momma joke to <nick>"""
+def yomomma(text, nick, conn, is_nick_valid):
+    """<nick> - Tells a yo momma joke to <nick>."""
     target = text.strip()
-    return '{}, {}'.format(target, random.choice(yo_momma).lower())
+    if not is_nick_valid(target) or target.lower() == conn.nick.lower():
+        target = nick
+    joke = random.choice(yo_momma).lower()
+    return '{}, {}'.format(target, joke)
 
 
 @hook.command(autohelp=False)
-def doit(message):
-    """- prints a do it line, example: mathmaticians do with a pencil"""
-    message(random.choice(do_it))
```

Comment:
Why replace the call to `message()` here? It would now hilight the user who ran the command where it didn't before.

Reply:
In my head I confused the `message` and `reply` parameter-functions so I was under the impression that returning a string called the `message` function anyway. I see now that returning a string calls `reply` which pings the user.
I will undo this…